parlai/core/torch_generator_agent.py

```diff
@@ -728,8 +715,21 @@ def _generate(self, batch, beam_size, max_ts):
             ).unsqueeze(-1)
             decoder_input = torch.cat([decoder_input, selection], dim=-1)
 
-        # get the top prediction for each
-        beam_preds_scores = [b.get_top_hyp() for b in beams]
+        # check that there is at least one finished candidate
+        # and assert that each of them contains only one EOS
+        n_best_beam_preds_scores = [b._get_rescored_finished() for b in beams]
```

Comment:
So we don’t want to call methods with a leading _, because they’re supposed to be protected. So can we upgrade the behavior of get_top_hyp instead of having this logic here?

Reply:
yeah, we get rid of get_top_hyp since we actually do not use it at all , since we do this EOS assert check for each candidate. Now get_rescored_finished will be not protected.